### PR TITLE
BW-compat poll refactor

### PR DESCRIPTION
**Pull request by canni**

### CODE REVIEW

**Comment by uktitools** [comment]
Can one of the admins verify this patch?

File: korben/bau/poll.py [hunk 1]
```diff
@@ -29,123 +29,145 @@ def get_entry_list(resp):
...
+    def __init__(self, client=None, against='ModifiedOn', comparator=operator.lt):
+        self.client = client or CDMSRestApi()
+        self.against = against
+        self.comparator = comparator
+
+        self.odata_metadata = services.db.get_odata_metadata()
+        self.django_metadata = services.db.get_django_metadata()
+        self.conn = services.db.poll_for_connection(config.database_odata_url)
+
+    def pool_entities(self, entities=None):
```
Comment (bmcorser) [inline korben/bau/poll.py]: Somewhere between reading the module name and opening the PR you lost track of the difference between "pool" and "poll".

Comment (bmcorser) [inline korben/bau/poll.py]: 🏊 vs ♻️

File: korben/bau/poll.py [hunk 2]
```diff
@@ -29,123 +29,145 @@ def get_entry_list(resp):
...
+            table = self.odata_metadata.tables[table_name]
+            col_names = [x.name for x in table.columns]
+
+            # assume a single column
+            primary_key = etl.utils.primary_key(table)
+            LOGGER.info('Starting reverse scrape for %s', table.name)
+
+            self.reverse_scrape(table, col_names, primary_key)
+
+    def reverse_scrape(self, table, col_names, primary_key):
```
Comment (bmcorser) [inline korben/bau/poll.py]: Docstrings!

File: korben/bau/poll.py [hunk 3]
```diff
@@ -29,123 +29,145 @@ def get_entry_list(resp):
...
-        table = odata_metadata.tables[table_name]
-        col_names = [x.name for x in table.columns]
-        # assume a single column
-        primary_key = etl.utils.primary_key(table)
-        LOGGER.info('Starting reverse scrape for %s', table.name)
-        reverse_scrape(
-            client, table, against, comparitor, col_names, primary_key, 0
-        )
+
+    pooler = CDMSPoller(client, against, comparitor)
```
Comment (bmcorser) [inline korben/bau/poll.py]: po**ol**er ≠ po**ll**er

File: korben/bau/poll.py [hunk 4]
```diff
@@ -29,123 +29,145 @@ def get_entry_list(resp):
...
+    pooler.pool_entities(entities)
 
 
 def main():
```
Comment (bmcorser) [inline korben/bau/poll.py]: The `poll` function is pointless now there's a class there. Get rid of it.

**Comment by bmcorser** [comment]
I'd like to get a Jenkins run here too, but I don't think the trigger is working as it should be.

@nusnewob ☝️

**Comment by sdonk** [comment]
LGTM

**Pull request merged**